src/components/controls/Controls.js

```diff
@@ -2,22 +2,22 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import styles from './Controls.css';
 
-const Controls = ({ actions, handleSelection }) => (
+const Controls = ({ dispatch }) => (
```

Comment:
so in here it has a dispatch, but later in your code I see you're mapping through the dispatch-- I think you should definitely re-name this to more accurately reflect the data type